src/main/webapp/js/finscholar/commonlistview/commonlistview.soy

```diff
@@ -51,4 +63,24 @@
 
 {template .endoflist}
   <p> This is the end of the list. </p>
+{/template}
+
+{template .sortByComponent}
+  {@param paramsToSortBy : map<string, string>}
+  <label for="sortBy">Choose Parameter to Sort By: </label>
+  <select name="sortBy" class="sort" id="sortBy" jsaction="change:commonlistview.change;">
+    {for $key in mapKeys($paramsToSortBy)}
+      <option value={$key}>{$paramsToSortBy[$key]}</option>
+    {/for}
+  </select>
+{/template}
+
+{template .sortOrderComponent}
+  {@param ascending : string}
+  {@param descending : string}
+  <label for="sortOrder">Choose Sort Order: </label>
```

Comment:
nit: same as above. "Order" would work just as well and unclutter the UI. Optionally a checkbox would with "Asc" as the label would do the trick.